core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1480,6 +1496,38 @@ private static Map<Symbol, Symbol> computeIdentityTranslations(Assignments assig
         return outputToInput;
     }
 
+    private int countRepartitionedRemoteExchangeNodes(PlanNode root)
+    {
+        return PlanNodeSearcher
+                .searchFrom(root)
+                .where(node -> node instanceof ExchangeNode exchangeNode && exchangeNode.getScope() == REMOTE && exchangeNode.getType() != REPLICATE)
```

Comment:
`exchangeNode.getType() != REPLICATE` -> `exchangeNode.getType() == PARTITIONED`

Reply:
You mean `exchangeNode.getType() == REPARTITION`?